wallet-new/docs/updates.md

```diff
@@ -0,0 +1,656 @@
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    V   |                                                |                                                |
+
+If we kept only one single field to store this data, such field might be sensible to data races and *ghost updates*,
+where we could accidentally override data in a way which would cause our internal invariants to be violated.
+
+The way updates will transition from the *next* to the *current* can be the following:
+
+- When a new update arrives, store it as the _next_ update. If there is already an
+  update designed as next, replace it.
+
+- When `GET /api/v1/update` is called:
+   - If the _current_ update is in the `Skipped` or `Applied` state, and a _next_ update is available,
+     then replace _current_ with _next_. *In-transit* `POST/PUT` http-requests won't
+     cause state inconsistencies by the virtue of the fact the input
+     `WalletSoftwareUpdate` won't match anymore.
+
+   - If the _current_ update is in the `Proposed` state, and a _next_ update is available,
+     then replace _current_ with _next_. *In-transit* `POST/PUT` http-requests won't
+     cause state inconsistencies by the virtue of the fact the input
+     `WalletSoftwareUpdate` won't match anymore.
+
+   - If the _current_ update is in the `Downloaded` state, and a _next_ update is available,
+     chances are we want to return the current (downloaded but not applied update) update so
+     that Daedalus can warn the user it has a downloaded (but not applied) update.
+
+- When `POST /api/v1/update` is called:
+    - If the input `WalletSoftwareUpdate` doesn't match the _current_ update, ignore the request.
+
+    - If the input `WalletSoftwareUpdate` does match the _current_ update, carry on as per
+      user story (4) and (4.1), then set the _current_ status to `Applied`.
+
+        - If a _next_ update is available, replace _next_ with _current_.
```

Comment:
I believe you wanted to say another way round -- replace _current_ with _next_.

Reply:
🤦‍♂️